cluster/event_monitor.go

```diff
@@ -38,10 +38,6 @@ func (em *EventsMonitor) Start(ec chan error) {
 					return
 				}
 			case err := <-errStream:
-				if err == io.EOF {
```

Comment:
It's not redundant. It's a bug. For some reason, docker engine would send `EOF` on event stream. Swarm manager needs to restart the event stream upon receiving EOF. See https://github.com/docker/swarm/blob/master/cluster/engine.go#L217. Here we cannot translate `EOF` to nil error.